xla/service/gpu/transforms/cudnn_fused_conv_rewriter.cc

```diff
@@ -576,33 +707,81 @@ void CaptureConvGraphRecursive(HloInstruction* instr,
       std::optional<PrimitiveType> f8_type = IsSaturatingCastToF8(users_user);
       if (f8_type.has_value()) {
         graph_string.ChangeDataType(f8_type.value());
-        num_linear_users++;
+        ++num_new_users;
         CaptureConvGraphRecursive(users_user, operands, aux_outputs,
-                                  graph_string, visited_instrs, final_instr);
+                                  graph_string, visited_instrs, final_instr,
+                                  num_endpoints);
         continue;
       }
       // Maximum of the absolute value (Amax) -- not a linear user
       if (Match(users_user,
-                m::Reduce(&op, m::Abs(m::Op(&operand0)), m::Op())) &&
-          AppliesMaxReduce(op)) {
-        if (graph_string.AppendOp("amax", op, {operand0})) {
-          aux_outputs.push_back(op);
-          num_nonlinear_users++;
+                m::Reduce(&op0, m::Abs(m::Op(&operand0)), m::Op())) &&
+          AppliesMaxReduce(op0)) {
+        if (graph_string.AppendOp("amax", op0, {operand0})) {
+          aux_outputs.push_back(op0);
+          ++num_new_users;
+        }
+        continue;
+      }
+      // Leaky ReLU, represented in the cuDNN graph as the maximum of the value
+      // and the slope times the minimum of zero and the value,
+      // max(x, alpha * min(0, x)).
+      if (Match(
+              users_user,
+              m::Select(
+                  &op2,
+                  m::Compare(
+                      &op0,
+                      m::Op().WithPredicate(unique_operand.CaptureOrVerifyFn()),
+                      m::Broadcast(&operand0, m::ConstantEffectiveScalar(0)))
+                      .WithComparisonDirection(ComparisonDirection::kGt),
+                  m::Op().WithPredicate(unique_operand.CaptureOrVerifyFn()),
+                  m::MultiplyAnyOrder(
+                      &op1,
+                      m::Op().WithPredicate(unique_operand.CaptureOrVerifyFn()),
+                      m::Broadcast(m::ConstantEffectiveScalar(&operand1)))))) {
+        if (!graph_string.OpInGraph(op0) && !graph_string.OpInGraph(op1) &&
+            !graph_string.OpInGraph(op2)) {
+          graph_string.AppendOp("min", op0, op2->shape().element_type(),
+                                {unique_operand.Instr(), operand0});
```

Comment:
As discussed offline, this "hack" of having cudnn ops that don't correspond to HLO ops should be commented. And you need to check the number of users of certain intermediate leaky_relu ops for correctness reasons, as discussed offline.